src/util/analytics-client.ts

```diff
@@ -1,85 +1,140 @@
 
+interface ScreenEventProps {
+	name: string;
+}
+
+interface TrackOpUiEventProps {
+	action: string;
+	actionSubject: string;
+	source: string;
+}
+
+const calculateClientEnv = () => {
+	switch (envVars.MICROS_ENVTYPE) {
+		case MicrosEnvTypeEnum.dev:
+			return "dev";
+		case MicrosEnvTypeEnum.staging:
+			return "stg";
+		case MicrosEnvTypeEnum.prod:
+			return "prod";
+	}
+	return "dev";
+};
+
+/**
+ *
+ * @param jiraHost
+ * @param eventType
+ * @param eventProps - even though it requires only a few fields leaving the rest as a loose Record, in fact all the properties
+ * 										 are well-defined and rigid. See the sample payloads here: https://bitbucket.org/atlassian/analytics-node-client/src/master/
+ * @param unsafeAttributes - free-form attributes, will be included as "attributes" property into the properties of the event
+ */
 export const sendAnalytics: {
-	(jiraHost: string, eventType: "screen", attributes: { name: string } & Record<string, unknown>);
-	(jiraHost: string, eventType: "track", attributes: { name: string, source: string } & Record<string, unknown>);
-	(jiraHost: string, eventType: "ui" | "operational", attributes: Record<string, unknown>);
-} = (jiraHost: string, eventType: string, attributes: Record<string, unknown> = {}): void => {
+	(jiraHost: string, eventType: "screen", eventProps: ScreenEventProps & Record<string, unknown>, attributes: Record<string, unknown>, accountId?: string);
+	(jiraHost: string, eventType: "ui" | "operational" | "track", eventProps: TrackOpUiEventProps  & Record<string, unknown>, attributes: Record<string, unknown>, accountId?: string);
+} = (jiraHost: string, eventType: string, eventProps: (ScreenEventProps | TrackOpUiEventProps)  & Record<string, unknown>, unsafeAttributes: Record<string, unknown> = {}, accountId?: string): void => {
 
 	logger.debug({ jiraHost }, analyticsClient ? "Found analytics client." : `No analytics client found.`);
 
-	if (!analyticsClient || !isNodeProd()) {
-		logger.warn("No analyticsClient or skipping sending analytics");
+	if (!analyticsClient) {
+		logger.warn("No analyticsClient available");
+		return;
+	}
+
+	if (isNodeTest()) {
+		logger.warn("Analytics is disabled in tests");
+		return;
+	}
+
+	if (isTestJiraHost(jiraHost) && envVars.MICROS_ENVTYPE === MicrosEnvTypeEnum.prod) {
+		logger.warn("Skip analytics for test jira in prod");
 		return;
 	}
 
 	if (!analyticsNodeClient){
 		// Values defined by DataPortal. Do not change their values as it will affect our metrics logs and dashboards.
 		analyticsNodeClient = analyticsClient({
-			env: "prod", // This needs to be "prod" as we're using prod Jira instances.
+			env: calculateClientEnv(),
 			product: "gitHubForJira"
 		});
 	}
 
-	const baseAttributes = {
-		userId: "anonymousId",
+	const baseEventProps = {
+		userId: accountId || "anonymousId",
 		userIdType: "atlassianAccount",
 		tenantIdType: "cloudId",
-		tenantId: "NONE"
+		tenantId: "NONE" // TODO: determine from jiraHost
```

Comment:
Or should we just put jiraHost here and shoot it to sqs, leave it to another service to calculate it?

Reply:
Yeah, that's the plan :)